File: src/main/java/cz/kamenitxan/jakon/webui/functions/PebbleExtension.java
package cz.kamenitxan.jakon.webui.functions;

import com.mitchellbosecke.pebble.extension.AbstractExtension;
import com.mitchellbosecke.pebble.extension.Function;
import cz.kamenitxan.jakon.webui.functions.GetAttributeFun;
import cz.kamenitxan.jakon.webui.functions.GetAttributeTypeFun;

import java.util.HashMap;
import java.util.Map;

/**
 * Created by tomaspavel on 6.10.16.
 */
public class PebbleExtension extends AbstractExtension {

	@Override
	public Map<String, Function> getFunctions() {
		return new HashMap<String, Function>() {{
			put("getAttr", new GetAttributeFun());
			put("getAttrType", new GetAttributeTypeFun());
			put("link", new LinkFun());
			put("i18n", new i18nFun());
		}};
	}
}

File: src/main/java/cz/kamenitxan/jakon/core/template/pebble/PebbleExtension.scala
package cz.kamenitxan.jakon.core.template.pebble

import java.util

import com.mitchellbosecke.pebble.extension.{AbstractExtension, Filter, Function}
import cz.kamenitxan.jakon.webui.functions.LinkFun

/**
  * Created by tomaspavel on 6.10.16.
  */
class PebbleExtension extends AbstractExtension {
	override def getFunctions: util.Map[String, Function] = {
		val extensions = new util.HashMap[String, Function]()
		extensions.put("i18n", new i18nFun)
		extensions.put("link", new LinkFun)
		extensions.put("value", new ValueFun)
		extensions
	}

	override def getFilters: util.Map[String, Filter] = {
		val extensions = new util.HashMap[String, Filter]()
		extensions.put("date", new JakonDateFilter)
		extensions.put("md", new MarkdownFilter)
		extensions
	}
}